### DISCUSSION TITLE
using pydoc-markdown to create code references

### DISCUSSION THREAD

**Discussion by younesStrittmatter** [AutoResearch/autora#383] [Core Contributions]
I played a bit with pydoc-markdown and maybe we can use that to create code-references for the packages (either in the packages themself or by importing them and then generating the markdown) As far as it also imports the math equations, so maybe it is worth to look into that.

**Comment by musslick** [comment]
What about our current way of generating code references? Is that, in principle, not compatible with the sub-package strategy?

**Reply by younesStrittmatter** [reply]
No, it doesn't work since the autora "mother" package doesn't include the src code (so we can not build from there), and in the child-packages the code reference are build in a way, that we can not import them (as far as I understood).

**Reply by younesStrittmatter** [reply]
But: For the time beeing I created a github-action here: https://github.com/younesStrittmatter/build-python-code-references-action and showcase the workflow for it here: https://github.com/younesStrittmatter/mkdocs-test-child/blob/main/src/test.py , that people can use to generate a static markdown version from the docstrings. Obviously it is not as professional as the mkdocs version, but I think it is very usefull yet for people that want to include the code-references in their documentation. I tested it on the "old" autora package and it seems to create all what we need.

**Reply by younesStrittmatter** [reply]
If people use it and encounter issues or have suggestions on how to enhance it, please feel free to contact me or create an issue in the https://github.com/younesStrittmatter/build-python-code-references-action repository.
